Libraries/LibWebView/EventLoop/EventLoopImplementationMacOS.mm

```diff
@@ -280,7 +320,10 @@ static void socket_notifier(CFSocketRef socket, CFSocketCallBackType notificatio
         break;
     }
 
-    CFSocketContext context { .version = 0, .info = nullptr, .retain = nullptr, .release = nullptr, .copyDescription = nullptr };
+    auto callback_context = adopt_ref(*new SocketNotifierCallbackContext);
+    callback_context->notifier = notifier.make_weak_ptr();
+    callback_context->ref();
```

Comment:
Are we sure the calls to `ref` and `unref` are balanced? It *looks* like we'll `ref` once more than `unref`, but I haven't run this locally - does AppKit invoke `release_socket_notifier_callback_context` one extra time?

Reply:
nope, this extra ref was a mistake. I got rid of it and now `SocketNotifierCallbackContext` destructor is actually invoked after unregistering.